code/game/objects/items/stacks/stack.dm

```diff
@@ -172,16 +196,16 @@
 		if(amount < R.req_amount * multiplier)
 			return
 
-		var/atom/O = new R.result_type(creation_loc)
+		var/atom/O
+		if(R.max_res_amount > 1) //Is it a stack?
+			O = new R.result_type(usr.drop_location(), R.res_amount * multiplier)
```

Comment:
You should probably replace all of the `creation_loc` stuff with `usr.drop_location()`